### PR TITLE
[SYCL] Add LIT tests for handler::host_task() (#3837)

### PR DESCRIPTION
**Pull request by v-klochkov**
This patch tests https://github.com/intel/llvm/pull/3837

Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

**Comment by v-klochkov** [comment]
Some tests check both handler::host_task() and handler::codeplay_host_task() to provide reasonable amount of testing for the deprecated codeplay_host_task() method.
Other tests did not get checks for codeplay_host_task() because a) there are no good reasons to think results would be different from host_task() as both are identical under the hood. 
Also, adding checks for both to all tests would only cause longer testing time. 

For example, the test 'SYCL/HostInteropTask/host-task-dependency3.cpp' runs about 37 secs on CPU and my initial move was to add checks for both methods there, which resulted into about 75 secs run on CPU, which seems not reasonable. Thus I simply replaced calls of codeplay_host_task() with host_task() there.

File: SYCL/Basic/host-task-dependency.cpp [hunk 1]
```diff
@@ -146,7 +155,7 @@ void test() {
   Context Ctx{{false}, Queue, {10}, {10}, {10}, {}, {}};
 
   // 0. setup: thread 1 T1: exec smth; thread 2 T2: waits; init flag F = false
-  auto A1 = std::async(std::launch::async, Thread1Fn, &Ctx);
+  auto A1 = std::async(std::launch::async, Thread1Fn<true>, &Ctx);
```
Comment (vladimirlaz) [inline SYCL/Basic/host-task-dependency.cpp]: Is it intended to be hardcoded and the next line is unspecified?

Comment (v-klochkov) [reply SYCL/Basic/host-task-dependency.cpp:158]: I am not sure what the question is about. Only 'Thread1Fn' was changed to a template function. 'Thread2Fn' was not changed same way.

Comment (v-klochkov) [reply SYCL/Basic/host-task-dependency.cpp]: Fixed in abca7580e61609a7ad2bdaeee9c00c9b7c9b885b. Thank you!

**Review by againull — APPROVED** [review]
LGTM

**Pull request merged**